### PR TITLE
Default to maximum allowed chunk size instead of chunk_size=0

### PR DESCRIPTION
**Pull request by lanice**
This implements changes discussed in #480. We disallow a chunk_size of 0, and have the default chunk_size being the maximum allowed chunk_size (max of `ChunkOffset` i.e. `uint32_t`).

I put the maximum allowed chunk size under Chunk::MAX_CHUNK_SIZE, though it also made sense to me having it under Table:: instead, so if anyone has thoughts on this, go ahead :)

The actual changes are in `Chunk`, `Table`, and `CsvMeta`. Most of the other changes are due to the fact that I had to change a lot of `load_table` calls with `chunk_size=0` to `MAX_CHUNK_SIZE`, mainly in the tests.

### CODE REVIEW

File: src/test/operators/import_csv_test.cpp [hunk 1]
```diff
@@ -155,12 +155,12 @@ TEST_F(OperatorsImportCsvTest, ChunkSize) {
 }
 
 TEST_F(OperatorsImportCsvTest, ChunkSizeZero) {
-  // chunk_size is defined as "20" in .meta file
-  auto importer = std::make_shared<ImportCsv>("src/test/csv/float_int_large_chunksize_0.csv");
+  // chunk_size is not defined in the meta file, resulting in the maximum allowed chunk size
+  auto importer = std::make_shared<ImportCsv>("src/test/csv/float_int_large_chunksize_max.csv");
   importer->execute();
 
   // check if chunk_size property is correct (maximum chunk size, 0 for unlimited)
```
Comment (Bensk1) [inline src/test/operators/import_csv_test.cpp]: Comment needs to be fixed.

**Comment by Bensk1** [comment]
What do you think about `chunk_capacity`?

**Comment by mrks** [comment]
"capacity" is used in a different way for data structures. For vectors, e.g., it denotes how much memory has been reserved.

**Comment by TwoLaid** [comment]
I gotta say, I still really like chunk_capacity

**Comment by mrks** [comment]
"Capacity" sounds like some type of limitation that is there because we can't fit any more in it. The maximum chunk size is there because we arbitrarily decided that we want to start a new chunk at a certain size.

I could live with it though.

**Comment by mrks** [comment]
@lanice Your call - either `chunk_capacity` or `max_chunk_size`. When you do the search/replace, please make sure that everyone who accesses it is in line with the new semantics.

**Comment by Bouncner** [comment]
I think if we ask 5 people what capacity or size means, we get 5 different answers.

We not go for something like `max_row_count`?

**Comment by mrks** [comment]
Works for me.

**Comment by Bensk1** [comment]
But we are talking about a member variable of `Table`, right? I believe `chunk` should be part of the name to clarify that this count/size/capacity concerns the chunks and not the table itself.

**Comment by lanice** [comment]
I like `max_chunk_size` more than `chunk_capacity`. That way we have `Table::max_chunk_size()` for the maximum chunk size, and `Chunk::size()` for the current individual size of each chunk.

The only thing I would then rename here is the constexpr I added, `Chunk::MAX_CHUNK_SIZE`, to `Chunk::MAX_ALLOWED_SIZE`. The 'chunk' in there is not really needed, since it is in the namespace anyway, so `Chunk::` should give it away what the `MAX_ALLOWED_SIZE` is for.

**Comment by klauck** [comment]
> The size is the number of elements in a container. The capacity is the maximum number of elements the container can currently hold.

I prefer `max_chunk_size`.

I do not really understand "ALLOWED" in `Chunk:MAX_ALLOWED_SIZE`.
I'm fine with `Chunk::MAX_SIZE`.
Another idea is to use `Chunk::MAX_OFFSET`, because this is where the limit comes from.

**Comment by lanice** [comment]
I thought the ALLOWED would help avoiding confusion between Table::max_chunk_size (which is the number that the user arbitrarily set), and the maximum possible chunk size, which is a fixed number.

**Comment by Bouncner** [comment]
I understand allowed it you explain it, but I wouldn't get it without your explanation. 

And I still think not talking about rows but about capacities/sizes is wrong. Just my 2ct.

**Comment by mrks** [comment]
`Chunk::MAX_SIZE` works for me as well. The fact that it's static and in capitals would be enough for me to show that it is a hard limit.

Using "size" for containers is common in the STL as well. Think about `std::vector::size()`. I would be fine with `Chunk::MAX_ROW_COUNT`, but how would we name the member of `Table`? I don't like `table.chunk_max_row_count` or `table.max_chunk_row_count` too much.

**Comment by Bouncner** [comment]
I see your point and I can absolutely live with any decision. Nonetheless, this thread showed that size() is still confusing.
And I don’t think max_chunk_row_count is too long.

**Comment by lanice** [comment]
Okay how about this. I changed the static constexpr to `Chunk::MAX_SIZE`. We merge it like it is now, to get this functionality in. And next time we meet for a fortnightly (I'll be in office again starting Tuesday), we can discuss if we want to rename one or both of these names, since then it is literally only a matter of search-and-replace.

**Pull request merged**